device/store.go

```diff
@@ -0,0 +1,113 @@
+// Copyright © 2020 Ispirata Srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package device
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/astarte-platform/astarte-go/interfaces"
+)
+
+type astarteMessageInfo struct {
+	StorageId      int    `gorm:"primaryKey;autoIncrement;not null"`
+	Retention      string `gorm:"not null"`
+	AbsoluteExpiry int64  `gorm:"not null"`
+	InterfaceName  string `gorm:"not null"`
+	Path           string `gorm:"not null"`
```

Comment:
This is something that bit us in the back in the past (see [here](https://github.com/astarte-platform/astarte-device-sdk-qt5/issues/16)): also add the interface major version to identify a message. The Rust SDK already [does so](https://github.com/astarte-platform/astarte-device-sdk-rust/blob/master/src/database.rs#L41) for the reason above.